Sources/BraintreeVenmo/BTVenmoClient.swift

```diff
@@ -313,23 +319,24 @@ import BraintreeCore
         apiClient.post("v1/payment_methods/venmo_accounts", parameters: parameters) { body, _, error in
             if let error = error as? NSError {
                 if error.code == BTCoreConstants.networkConnectionLostCode {
-                    self.apiClient.sendAnalyticsEvent("ios.pay-with-venmo.network-connection.failure")
+                    self.apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeNetworkConnectionLost)
                 }
 
-                self.apiClient.sendAnalyticsEvent("ios.pay-with-venmo.vault.failure")
+                self.apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeFailed)
                 self.appSwitchCompletion(nil, error)
                 return
             }
 
             guard let body else {
+                self.apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeFailed)
                 self.appSwitchCompletion(nil, BTVenmoError.invalidBodyReturned)
                 return
             }
 
             let venmoAccountJSON: BTJSON = body["venmoAccounts"][0]
             let venmoAccountNonce: BTVenmoAccountNonce = BTVenmoAccountNonce.venmoAccount(with: venmoAccountJSON)
 
-            self.apiClient.sendAnalyticsEvent("ios.pay-with-venmo.vault.success")
+            self.apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeSucceeded)
```

Comment:
The completion below sends either a nonce or an error if present - we should check `venmoAccountJSON.asError()` first and send the `tokenizeFailed` event there with a nil nonce in the completion. Right now we'd send a `tokenizeSucceeded` event even if there is an error. I think we did a similar check in Card.

Reply:
Great catch! I missed that it would be sending an error from checking the JSON body.

Reply:
Nope, we don't care about those - only the ones on the doc are the ones we want to retain